### PR TITLE
Jw fix diff 4

### PR DESCRIPTION
**Pull request by id4ho**
###  Summary

Generate a default alt text for an image if one isn't provided. https://jira.tinyspeck.com/browse/PSF-808

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackhq/htmlsanitizer-hack/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've written tests to cover the new code and functionality included in this PR.

### CODE REVIEW

File: src/AttrTransform/ImgRequired.hack [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// must be called POST validation
```
Comment (id4ho) [inline src/AttrTransform/ImgRequired.hack:1]: https://github.com/ezyang/htmlpurifier/blob/master/library/HTMLPurifier/AttrTransform/ImgRequired.php is original PHP

File: src/Definition/HTMLDefinition.hack [hunk 2]
```diff
@@ -425,9 +425,22 @@ class HTMLPurifier_HTMLDefinition extends HTMLPurifier\HTMLPurifier_Definition {
...
             "sizes" => new AttrDef\HTMLPurifier_AttrDef_Text()
         ];
-        $img_element = new HTMLPurifier\HTMLPurifier_ElementDef(true, $img_add_attr, vec[], vec[], vec[], 
-            new ChildDef\HTMLPurifier_ChildDef_Empty(), null, '', false, vec["alt", "src"], vec[],
-            vec[], '', false);
+        $img_element = new HTMLPurifier\HTMLPurifier_ElementDef(
+          true,
+          $img_add_attr,
+          vec[],
+          vec[new AttrTransform\HTMLPurifier_AttrTransform_ImgRequired()],
```
Comment (id4ho) [inline src/Definition/HTMLDefinition.hack:432]: pre_attribute_transforms

File: src/Definition/HTMLDefinition.hack [hunk 3]
```diff
@@ -425,9 +425,22 @@ class HTMLPurifier_HTMLDefinition extends HTMLPurifier\HTMLPurifier_Definition {
...
+          vec[new AttrTransform\HTMLPurifier_AttrTransform_ImgRequired()],
```
Comment (id4ho) [inline src/Definition/HTMLDefinition.hack:433]: post_attribute_transforms

**Review by cgao-slack — APPROVED** [review]
Thanks for fixing this.

**Review by og-slack — APPROVED** [review]
🚀

**Pull request merged**